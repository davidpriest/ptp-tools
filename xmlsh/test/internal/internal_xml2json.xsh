# test of xml2json

F=../../samples/data/books.jxml
xml2json -p < $F

