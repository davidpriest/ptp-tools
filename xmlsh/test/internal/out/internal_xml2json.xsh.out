
 {
  "BOOKLIST":
   {
    "_children":
     [
      {
       "BOOKS":
        {
         "_children":
          [
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"MMP"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["Pride and Prejudice"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Jane Austen"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Modern Library"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["2002-12-31"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["4.95"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["187"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0679601686"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["352"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"in"
                    },
                   "_children":
                    ["8.3 5.7 1.1"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"oz"
                    },
                   "_children":
                    ["6.1"]
                  }
                }]
             }
           },
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"P"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["Wuthering Heights"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Charlotte Brontï¿½"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Penguin Classics"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["2002-12-31"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["6.58"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["113"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0141439556"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["430"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"in"
                    },
                   "_children":
                    ["1.0 5.2 7.8"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"oz"
                    },
                   "_children":
                    ["11.2"]
                  }
                }]
             }
           },
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"P"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["Tess of the d'Urbervilles"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Thomas Hardy"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Bantam Classics"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["1984-05-01"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["4.95"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["85"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0553211684"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["480"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"in"
                    },
                   "_children":
                    ["6.8 4.2 0.8"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"oz"
                    },
                   "_children":
                    ["7.7"]
                  }
                }]
             }
           },
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"P"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["Jude the Obscure"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Thomas Hardy"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Penguin Classics"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["1998-09-01"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["4.95"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["129"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0140435387"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["528"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"in"
                    },
                   "_children":
                    ["7.8 5.2 0.9"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"oz"
                    },
                   "_children":
                    ["10.9"]
                  }
                }]
             }
           },
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"H"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["The Big Over Easy"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Jasper Fforde"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Hodder & Stoughton"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["2005-07-11"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["16.47"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["129"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0340835672"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["346"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"cm"
                    },
                   "_children":
                    ["22.5 18.0 3.5"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"g"
                    },
                   "_children":
                    ["390"]
                  }
                }]
             }
           },
           {
            "ITEM":
             {
              "_attributes":
               {
                "CAT":"P"
               },
              "_children":
               [
                {
                 "TITLE":
                  {
                   "_children":
                    ["The Eyre Affair"]
                  }
                },
                {
                 "AUTHOR":
                  {
                   "_children":
                    ["Jasper Fforde"]
                  }
                },
                {
                 "PUBLISHER":
                  {
                   "_children":
                    ["Penguin"]
                  }
                },
                {
                 "PUB-DATE":
                  {
                   "_children":
                    ["2003-02-25"]
                  }
                },
                {
                 "LANGUAGE":
                  {
                   "_children":
                    ["English"]
                  }
                },
                {
                 "PRICE":
                  {
                   "_children":
                    ["16.47"]
                  }
                },
                {
                 "QUANTITY":
                  {
                   "_children":
                    ["129"]
                  }
                },
                {
                 "ISBN":
                  {
                   "_children":
                    ["0142001805"]
                  }
                },
                {
                 "PAGES":
                  {
                   "_children":
                    ["384"]
                  }
                },
                {
                 "DIMENSIONS":
                  {
                   "_attributes":
                    {
                     "UNIT":"in"
                    },
                   "_children":
                    ["7.8 5.0 0.9"]
                  }
                },
                {
                 "WEIGHT":
                  {
                   "_attributes":
                    {
                     "UNIT":"oz"
                    },
                   "_children":
                    ["9.0"]
                  }
                }]
             }
           }]
        }
      },
      {
       "CATEGORIES":
        {
         "_attributes":
          {
           "DESC":"Miscellaneous categories"
          },
         "_children":
          [
           {
            "CATEGORY":
             {
              "_attributes":
               {
                "CODE":"P",
                "DESC":"Paperback"
               }
             }
           },
           {
            "CATEGORY":
             {
              "_attributes":
               {
                "CODE":"MMP",
                "DESC":"Mass-market Paperback"
               }
             }
           },
           {
            "CATEGORY":
             {
              "_attributes":
               {
                "CODE":"H",
                "DESC":"Hard Cover"
               }
             }
           }]
        }
      }]
   }
 }